base/float.jl

```diff
@@ -502,6 +502,18 @@ for Ti in (Int64,UInt64,Int128,UInt128)
     end
 end
 
+for Ti in (Int64,UInt64,Int128,UInt128)
+    @eval begin
+        ==(x::Float16, y::$Ti) = Float64(x)==Float64(y)
+        ==(x::$Ti, y::Float16) = Float64(x)==Float64(y)
+
+        <(x::Float16, y::$Ti) = Float64(x)<Float64(y)
+        <(x::$Ti, y::Float16) = Float64(x)<Float64(y)
+
+        <=(x::Float16, y::$Ti) = Float64(x)<=Float64(y)
+        <=(x::$Ti, y::Float16) = Float64(x)<=Float64(y)
+    end
+end
```

Comment:
`Inf16 == typemax(UInt16)` is also currently incorrect. We could fix it with methods like this that promote to `Float32`.